autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -264,6 +260,20 @@ def transform(
 
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = cast(Type[pd.DataFrame], X)
+            if self.all_nan_columns is not None:
+                for column in X.columns:
+                    if column in self.all_nan_columns:
+                        if not X[column].isna().all():
+                            X[column] = np.nan
+                        X[column] = pd.to_numeric(X[column])
+            if len(self.categorical_columns) > 0:
+                if self.column_transformer is None:
+                    raise AttributeError("Expect column transformer to be built"
+                                         "if there are categorical columns")
+                categorical_columns = self.column_transformer.transformers_[0][-1]
+                for column in categorical_columns:
+                    if X[column].isna().all():
+                        X[column] = X[column].astype('object')
```

Comment:
This is quicker

```suggestion
                all_nan_columns = set() self.all_nan_columns is None else self.all_nan_columns
                for column in categorical_columns:
                    if column in all_nan_columns or X[column].isna().all():
                        X[column] = X[column].astype('object')
```

Reply:
actually, this is the `transform` method. and `self.all_nan_columns` is built when we fit on the training dataset. So, basically the validation data or the test data all nan columns are not present in `self.all_nan_columns` and hence we need this check.

Reply:
Yeah, I knew it, but wouldn't we like to change all the columns if it has any possibilities that it has nan?

Btw, this part also needs to be changed for a huge number of features. (I will send another comment in the latest change)